gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/jdbc/EmbedStatement.java

```diff
@@ -1202,6 +1203,11 @@ else if (SanityManager.DEBUG) {
               execFlags = (short) GemFireXDUtils.set(execFlags,
                   GenericStatement.QUERY_HDFS, true);
             }
+
+            if (Misc.getMemStore().isSnappyStore() && routeQueryEnabled(cc)) {
```

Comment:
isSnappyStore check is redundant since that should be covered in  routeQueryEnabled

Reply:
Yes isSnappyStore is redundant for Snappy and saves the the function call for rowstore mode. Maybe we can remove this check.